pkg/rtc/participant.go

```diff
@@ -2055,3 +2057,16 @@ func (p *ParticipantImpl) GetCachedDownTrack(trackID livekit.TrackID) (*webrtc.R
 
 	return nil, sfu.ForwarderState{}
 }
+
+func (p *ParticipantImpl) handleNegotiationFailed() {
+	p.params.Logger.Infow("negotiation failed, notify client do full reconnect")
+	_ = p.writeMessage(&livekit.SignalResponse{
```

Comment:
if negotiation failed caused by websocket failed,  the Leave message might not arrive client

Reply:
I think that should be fine as client will detect that.

Question: should we close the signal connection below? Or should reconnect handle that?

Reply:
Have no idea which is better (close or left it open). Just concern if client received Leave request, then subsequent message will confuse it. @davidzhao any suggestion about this?

Reply:
as soon as clients receive the leave message, it'll ignore subsequent events. it's probably safer to close it right away.